tests/ship_client.cpp

```diff
@@ -58,32 +57,29 @@ int main(int argc, char* argv[]) {
       statehistory_server = socket_address.substr(socket_address.find("unix://") + strlen("unix://") + 1);
    } else {
       std::string::size_type colon = socket_address.find(':');
-      eosio::check(colon != std::string::npos, "Missing ':' seperator in Websocket address and port");
+      FC_ASSERT(colon != std::string::npos, "Missing ':' seperator in Websocket address and port");
       statehistory_server = socket_address.substr(0, colon);
       statehistory_port = socket_address.substr(colon + 1);
    }
 
    std::cerr << "[\n{\n   \"status\": \"construct\",\n   \"time\": " << time(NULL) << "\n},\n";
 
    try {
-      auto run = [&](auto& stream) { // C++20: [&]<typename SocketStream>(SocketStream& stream)
+      auto run = [&]<typename SocketStream>(SocketStream& stream) {
          {
             boost::beast::flat_buffer abi_buffer;
             stream.read(abi_buffer);
-            std::string abi_string((const char*)abi_buffer.data().data(),
-                                   abi_buffer.data().size());
-            eosio::json_token_stream token_stream(abi_string.data());
-            eosio::abi_def abidef =
-                eosio::from_json<eosio::abi_def>(token_stream);
-            eosio::convert(abidef, abi);
+            std::string abi_string = boost::beast::buffers_to_string(abi_buffer.data());
+            //remove all tables since their names are invalid; tables not needed for this test
+            std::regex scrub_all_tables(R"(\{ "name": "[^"]+", "type": "[^"]+", "key_names": \[[^\]]*\] \},?)");
+            abi_string = std::regex_replace(abi_string, scrub_all_tables, "");
```

Comment:
Why not use `ship_abi_without_tables()` here? You will need to link to state_history, but that seems ok.

Reply:
An important aspect of the test is consuming the ABI from the websocket. `ship_abi_without_tables()` will skip that.

Reply:
Oh right. Maybe have a `ship_abi_without_tables(const std::string& abi_string)` ?

Reply:
Yeah I thought about that but where to put it? I liked not linking to state_history because it clearly indicates the full isolation from anything "libship" related. I could always make a new tiny lib just for it. Though, ship_client and ship_streamer already seem to be quite a bit of a copy paste in areas so eh..

Reply:
Sounds fine, just leave as is.